content/php/concepts/string-functions/terms/sprintf/sprintf.md

```diff
@@ -0,0 +1,102 @@
+The following table lists the possible specifiers:
+
+| Specifier | Description                                                                                         |
+| --------: | --------------------------------------------------------------------------------------------------- |
+|       `b` | Numeric value is converted to an integer and shown as a binary number.                              |
+|       `c` | Numeric value is converted to an integer and is shown as a character with that ASCII value.         |
+|       `d` | Numeric value is converted to an integer and is shown as a signed decimal number.                   |
+|       `e` | Numeric value is converted to a float and shown with scientific notation with a lower case "e".     |
+|       `E` | Numeric value is converted to a float and shown with scientific notation with an upper case "E".    |
+|       `f` | Numeric value is converted to a float and shown as a locale aware floating-point number.            |
+|       `F` | Numeric value is converted to a float and shown as a non-locale aware floating-point number.        |
+|       `g` | General numeric format converting numeric value to a float and treating exponents as `e`.           |
+|       `G` | General numeric format converting numeric value to a float and treating exponents as `E`.           |
+|       `h` | Like `g` but treats floats as `F`.                                                                  |
+|       `H` | Like `G` but treats floats as `F`.                                                                  |
+|       `o` | Numeric value is converted to an integer and shown as an octal number.                              |
+|       `s` | Value is treated as a string.                                                                       |
+|       `u` | Numeric value is converted to an integer and is shown as a unsigned decimal number.                 |
+|       `x` | Numeric value is converted to an integer and shown as a hexadecimal number with lower case letters. |
+|       `X` | Numeric value is converted to an integer and shown as a hexadecimal number with upper case letters. |
+
+## Example
+
+```php
+<?php
+$str1 = 'PHP';
+$str2 = 'code';
+$format = "This string is right justified and padded with \"^\" characters: \"%'^8s\".  \nThis string is left justified and padded with spaces: \"%-8s\".";
+echo sprintf($format, $str1, $str2);
+?>
+```
+
+This results in the following output:
+
+```shell
+This string is right justified and padded with "^" characters: "^^^^^PHP".
+This string is left justified and padded with spaces: "code    ".
+```
+
+## Codebyte Example
+
+```codebyte/php
+<?php
+$num1 = 63;
+$num2 = 87;
+$format = 'The number %1$d in hexadecimal is %1$04X. The number %2$d in binary is %2$08b.';
+echo sprintf($format, $num1, $num2);
+?>
+```
+
+**Note:** The format string must be in single quotes or else PHP will try to interpret the `$d` in the `%1$d` and `%2$d` format directives as a variable.
```

Comment:
```suggestion
**Note**: The format string must be in single quotes, otherwise PHP will try to interpret the `$d` in the `%1$d` and `%2$d` format directives as a variable.
```

otherwise, imo objectively, reads better than or else